partialWaveFit/fitResult.cc

```diff
@@ -998,3 +1000,151 @@ fitResult::intensityErr(const string& waveNamePattern) const
+	std::set<rpwa::multibinBoundariesType> allMultibinBoundaries;
+	std::map<rpwa::multibinBoundariesType, double> negLogLikeOfBestResult;
+	std::map<rpwa::multibinBoundariesType, std::pair<size_t, int> > fileEntryOfBestResult;
+	std::map<rpwa::multibinBoundariesType, std::list<rpwa::fitResult>::iterator > iterOfBestResult;
+	std::map<rpwa::multibinBoundariesType, double> negLogLikeOfBestConvergedResult;
+	std::map<rpwa::multibinBoundariesType, std::pair<size_t, int> > fileEntryOfBestConvergedResult;
+	std::map<rpwa::multibinBoundariesType, std::list<rpwa::fitResult>::iterator > iterOfBestConvergedResult;
+	for (size_t i = 0; i < trees.size(); ++i) {
+		printInfo << "load fit results from file '" << fileNames[i] << "'" << std::endl;
+
+		TTree* tree = trees[i];
+		for (int j = 0; j < tree->GetEntries(); ++j) {
+			tree->GetEntry(j);
+			if( onlyConvergedResults and not fitResultIn->converged())
+				continue;
+
+			const rpwa::multibinBoundariesType& multibinBoundaries = fitResultIn->multibinBoundaries();
+			allMultibinBoundaries.insert(multibinBoundaries);
+			std::list<rpwa::fitResult>::iterator it;
+
+			if (not onlyBestInMultibin) {
+				if (stripMatricesFromFurtherAttempts) {
+					fitResultIn->stripCovMatrix();
+					fitResultIn->stripIntegralMatrices();
+				}
+				it = fitResultsInMultibins[multibinBoundaries].insert(fitResultsInMultibins[multibinBoundaries].end(), *fitResultIn);
+
+				if(fitResultIn->converged()){
+					if (negLogLikeOfBestConvergedResult.find(multibinBoundaries) == negLogLikeOfBestConvergedResult.end() or // first result of this bin
+					    fitResultIn->logLikelihood() < negLogLikeOfBestConvergedResult[multibinBoundaries]) { // found new best result
+						negLogLikeOfBestConvergedResult[multibinBoundaries] = fitResultIn->logLikelihood();
+						fileEntryOfBestConvergedResult[multibinBoundaries] = std::make_pair(i, j);
+						iterOfBestConvergedResult[multibinBoundaries] = it;
+					}
+				}
+			}
+			if (negLogLikeOfBestResult.find(multibinBoundaries) == negLogLikeOfBestResult.end() or // first result of this bin
+			    fitResultIn->logLikelihood() < negLogLikeOfBestResult[multibinBoundaries]) { // found new best result
+				negLogLikeOfBestResult[multibinBoundaries] = fitResultIn->logLikelihood();
+				fileEntryOfBestResult[multibinBoundaries] = std::make_pair(i, j);
+				iterOfBestResult[multibinBoundaries] = it;
+			}
+		}
+	}
+
+
+	// finally add the result with the best likelihood, if not yet stored
+	if (onlyBestInMultibin) {
+		for (const auto& multibinBoundaries : allMultibinBoundaries) {
+			// load best result
+			trees[fileEntryOfBestResult[multibinBoundaries].first]->GetEntry(fileEntryOfBestResult[multibinBoundaries].second);
+			std::list<fitResult>& fitResults = fitResultsInMultibins[multibinBoundaries];
+			assert(fitResults.size() == 0);
+			fitResults.push_back(*fitResultIn);
+		}
+	} else if (stripMatricesFromFurtherAttempts) {
```

Comment:
In principle, I agree.
From my point of view, there are two possibilities to realize this and both have serious drawbacks.
Either you first load all fit results with their matrices into memory, than determine the best one and finally strip the matrices from all other results. However, this would mean that you need enough memory to keep all fit results with their matrices of all attempts from all multibins (if you want collect your results e.g.) in memory, which might be quite a lot.
Or you have to go twice over all fit result files. Fist to determine the best results and second to load the results, while keeping the matrices only for the best results. Going twice over all files is also not really a good solution.